aws_lambda_builders/actions.py

```diff
@@ -138,24 +144,48 @@ def execute(self):
             utils.create_symlink_or_copy(str(source_path), str(destination_path))
 
 
+class LinkSinglePathAction(BaseAction):
+    NAME = "LinkSource"
+
+    DESCRIPTION = "Creates symbolic link at dest, pointing to source"
+
+    PURPOSE = Purpose.LINK_SOURCE
+
+    def __init__(self, source: Union[str, os.PathLike], dest: Union[str, os.PathLike]):
+        self._source = source
+        self._dest = dest
+
+    def execute(self):
+        destination_path = Path(self._dest)
+        if not destination_path.exists():
+            os.makedirs(destination_path.parent, exist_ok=True)
+        utils.create_symlink_or_copy(str(self._source), str(destination_path))
+
+
 class CopyDependenciesAction(BaseAction):
     NAME = "CopyDependencies"
 
     DESCRIPTION = "Copying dependencies while skipping source file"
 
     PURPOSE = Purpose.COPY_DEPENDENCIES
 
-    def __init__(self, source_dir, artifact_dir, destination_dir):
+    def __init__(self, source_dir, artifact_dir, destination_dir, maintain_symlinks=False):
         self.source_dir = source_dir
         self.artifact_dir = artifact_dir
         self.dest_dir = destination_dir
+        self.maintain_symlinks = maintain_symlinks
 
     def execute(self):
         deps_manager = DependencyManager(self.source_dir, self.artifact_dir, self.dest_dir)
 
         for dependencies_source, new_destination in deps_manager.yield_source_dest():
-            if os.path.isdir(dependencies_source):
-                copytree(dependencies_source, new_destination)
+            if os.path.islink(dependencies_source) and self.maintain_symlinks:
+                os.makedirs(os.path.dirname(new_destination), exist_ok=True)
+                linkto = os.readlink(dependencies_source)
+                create_symlink_or_copy(linkto, new_destination)
+                shutil.copystat(dependencies_source, new_destination, follow_symlinks=False)
```

Comment:
```
create_symlink_or_copy(linkto, new_destination)
shutil.copystat(dependencies_source, new_destination, follow_symlinks=False)
```

We are symlinking and copying?

Reply:
same question here ... may be I am missing something, but will the copy ovwerwrite the created symlink ?

Reply:
copystat copies the metadata: https://docs.python.org/3/library/shutil.html#shutil.copystat

Since we are "copying" the symlink by creating a new one that points to the same location as the other one, I thought we might want to copy over the metadata. Let me know if you think this is not necessary.